client/src/store/Filetypes.ts

```diff
@@ -0,0 +1,31 @@
+import { Module } from 'vuex';
+import { getValidFileTypes, Types } from '@/lib/api/viame.service';
+
+export interface FiletypeState {
+    filetypes: null | Record<string, Types>;
+}
+
+const filetypeModule: Module<FiletypeState, never> = {
+  namespaced: true,
+  state: {
+    filetypes: null,
+  },
+  mutations: {
+    setFiletypes(state, filetypes) {
+      state.filetypes = filetypes;
+    },
+  },
+  actions: {
+    async fetchFiletypes({ commit, state }) {
+      if (state.filetypes === null) {
+        const { data } = await getValidFileTypes();
+
+        commit('setFiletypes', data);
+        return data;
```

Comment:
No need to return early here.   Easier to guarantee consistent behavior with fewer code paths.

Reply:
I think that unless there's a reason to keep it here, that we should remove the return statements altogether. We're not currently saving the return value of this function anyway, and keeping the state as the source of truth seems like a good practice to me.